packages/SwingSet/src/kernel/slogger.js

```diff
@@ -1,120 +1,127 @@
 import { q } from '@endo/errors';
+import { makeLimitedConsole } from '@agoric/internal/src/ses-utils.js';
+
+/** @import {LimitedConsole} from '@agoric/internal/src/js-utils.js'; */
 
 const IDLE = 'idle';
 const STARTUP = 'startup';
 const DELIVERY = 'delivery';
 
-function makeCallbackRegistry(callbacks) {
-  const todo = new Set(Object.keys(callbacks));
+const noopFinisher = harden(() => {});
+
+/** @typedef {(...finishArgs: unknown[]) => unknown} AnyFinisher */
+/** @typedef {Partial<Record<Exclude<keyof KernelSlog, 'write'>, (methodName: string, args: unknown[], finisher: AnyFinisher) => unknown>>} SlogWrappers */
```

Comment:
Would it be worth indicating that each SlogWrapper method can return a finisher, even though that information would be subsumed in many places?
```suggestion
/** @typedef {Partial<Record<Exclude<keyof KernelSlog, 'write'>, (methodName: string, args: unknown[], finisher: AnyFinisher) => (AnyFinisher | unknown)>>} SlogWrappers */
```